src/model_config.cc

```diff
@@ -414,12 +418,22 @@ ProtocolStringToDataType(const char* dtype, size_t len)
       return inference::DataType::TYPE_FP64;
     }
   } else if (*dtype == 'B') {
-    if (dtype[1] == 'Y') {
-      if (!strcmp(dtype + 2, "TES")) {
-        return inference::DataType::TYPE_STRING;
-      }
-    } else if (!strcmp(dtype + 1, "OOL")) {
-      return inference::DataType::TYPE_BOOL;
+    switch (dtype[1]) {
```

Comment:
@GuanLuo This file seems mostly duplicated from [core/src/model_config.cc](https://github.com/triton-inference-server/core/blob/32c944f84f8d2952f6631a34d1139c6ef93af519/src/model_config.cc) after this [PR](https://github.com/triton-inference-server/common/pull/48) - is this intended? or do we plan to remove the core version? I notice there are some functions not included in both.

Reply:
The PR is for moving generic utils in `model_config.h/.cc` from core repo to common repo, the missing functions are core specific and moved to `model_config_utils.h/.cc`. There is still a copy `model_config.cc` in core but it seems to be leftover from the move, it is not being used anywhere

Reply:
Filed DLIS-3781 to follow-up. LGTM